hammers/query.py

```diff
@@ -153,11 +153,35 @@ def owned_ips(db, project_ids):
     '''.format(projcol=project_col(db.version))
     return db.query(sql, args=[project_ids], limit=None)
 
+@query
+def floating_ips_to_leases(db, floating_ip_ids):
+    '''Return 'active' leases from a tuple of floating ip ids.'''
+    sql = '''
+    SELECT bl.id AS lease_id
+        , bl.action AS action
+        , bl.end_date AS end_date
+        , bl.deleted_at AS deleted_at
+	    , nfi.id AS ip_id
+    FROM neutron.floatingips nfi
+    LEFT JOIN neutron.ports np ON nfi.fixed_port_id=np.id
+    LEFT JOIN nova.instances ni ON np.device_id=ni.uuid
+    LEFT JOIN ironic.nodes ino ON ni.uuid=ino.instance_uuid
+    LEFT JOIN blazar.computehosts bc ON ino.uuid = bc.hypervisor_hostname
+    LEFT JOIN blazar.computehost_allocations bca ON bca.compute_host_id=bc.id
+    LEFT JOIN blazar.reservations br ON bca.reservation_id=br.id
+    LEFT JOIN blazar.leases bl ON br.lease_id=bl.id
+    WHERE bl.project_id=nfi.project_id
+        AND bl.deleted_at is NULL
+        AND nfi.id IN {floating_ip_ids};
+    '''.format(floating_ip_ids=str(floating_ip_ids))
```

Comment:
This is susceptible to SQL injection. Using the `IN` is a bit tricky, but possible to do like this:

```python
floating_ips_varargs = ','.join(['%s'] * len(floating_ip_ids))
sql = '''
  ...
  WHERE ...
    AND nfi.id IN ({ floating_ips_varargs })
'''.format(floating_ips_varargs=floating_ips_varargs)
return db.query(sql, floating_ip_ids, limit=None)
```
Example source: https://stackoverflow.com/a/589416

The approach here is to generate a query using token placeholders, one for each value you will be replacing. So it will generate a statement like `WHERE IN (%s, %s, %s)` if there are 3 being passed in. Then those tokens are replaced by the MySQL library itself. This is the safest way as it properly escapes all input.